example/cmd/microd/main.go

```diff
@@ -81,6 +81,18 @@ func (c *cmdDaemon) Run(cmd *cobra.Command, args []string) error {
 			return nil
 		},
 
+		PreBootstrap: func(s *state.State, initConfig map[string]string) error {
```

Comment:
Just for my understanding. I guess when you import `MicroCluster` you can overwrite those hooks with your own ones?

Reply:
Yes, you can specify your own set of hooks. For example with MicroCloud we use `PostBootstrap` and `PostJoin` to stop the mDNS broadcast of our availability, since we just finished bootstrapping or joining a cluster:

https://github.com/canonical/microcloud/blob/82613aa8398252af24f27eb26109b185531cefaa/microcloud/service/microcloud.go#L57-L63